### PR TITLE
Don't reset timers unnecessarily

### PR DESCRIPTION
**Pull request by jhiesey**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain:

**What changes did you make? (Give an overview)**
If a timeout timer expiration will only be changed by 5% of the
timeout time, don't change it. This eliminates a large number of
slow setTimeout/clearTimeout calls.

**Which issue (if any) does this pull request address?**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -597,22 +599,34 @@ class Wire extends stream.Duplex {
...
-    this._timeout = null
-  }
+    if (!this._timeoutMs) {
+      return
+    }
+
+    const timeoutExpiresAt = Date.now() + this._timeoutMs
 
-  _updateTimeout () {
-    if (!this._timeoutMs || !this.requests.length || this._timeout) return
```
Comment (feross) [inline index.js:614]: Is it okay that `this.requests.length` is not being checked now?

**Review by feross — APPROVED** [review]

**Comment by feross** [comment]
Looks like there were some test failures. Maybe due to the comment I left?

**Comment by jhiesey** [comment]
Test failures should be fixed!

**Review by feross — APPROVED** [review]

**Comment by feross** [comment]
3.3.0

**Pull request merged**